node/pkg/gwrelayer/gwrelayer.go

```diff
@@ -171,15 +267,33 @@ func shouldPublish(payload []byte, targetChain vaa.ChainID, targetAddress vaa.Ad
 	return true, nil
 }
 
+// shouldPublishToTokenBridge returns true if a message should be forwarded to the token bridge, false if not.
+func shouldPublishToTokenBridge(tokenBridges tokenBridgeMap, v *vaa.VAA) bool {
+	if _, exists := tokenBridges[tokenBridgeKey{emitterChainId: v.EmitterChain, emitterAddr: v.EmitterAddress}]; !exists {
+		return false
+	}
+
+	if len(v.Payload) == 0 {
+		return false
+	}
+
+	// We only forward attestations (type two).
```

Comment:
type two covers simple (non-payload3) token bridge messages as well, right?

Should we lock this down further to **only** forward attestations? I actually think it's okay to also forward non-payload3 messages, since people may accidentally initiate transfers directly to Gateway. cc @evan-gray

Reply:
No, type two is only attestations:

https://github.com/wormhole-foundation/wormhole/blob/main/whitepapers/0003_token_bridge.md?plain=1#L185